lib/reader.rb

```diff
@@ -0,0 +1,33 @@
+require 'json'
+
+class Reader
+
+  def initialize(path, file = nil)
+    @path = path
+    @file = file
+  end
+
+  def read_contacts
+    file.nil? ? parse(read) : parse(tests_read)
+  end
+
+  private
+
+  attr_reader :path, :file
+
+  def parse(contacts)
+    contacts == '' ? [] : JSON.parse(contacts)
+  end
+
+  def read
+    file = File.open(path, "r")
```

Comment:
You can also do:
```
File.open(path, "r") do |file|
  file.read
end
```
which should handle closing the file for you. Maybe that helps with your test?